src/main/java/org/zalando/nakadi/service/AuthorizationValidator.java

```diff
@@ -185,4 +199,43 @@ public void validateAuthorization(final EventType original, final EventTypeBase
         validateAuthorization(newAuth);
     }
 
+    public AdminAuthorization getAdmins() {
+        final List<Permission> allPermissions = authorizationDbRepository.listAdmins();
+        final List<AuthorizationAttribute> adminPermissions = allPermissions.stream()
+                .filter(p -> p.getOperation() == AuthorizationService.Operation.ADMIN)
+                .map(p -> p.getAuthorizationAttribute())
+                .collect(Collectors.toList());
+        final List<AuthorizationAttribute> readPermissions = allPermissions.stream()
+                .filter(p -> p.getOperation() == AuthorizationService.Operation.READ)
+                .map(p -> p.getAuthorizationAttribute())
+                .collect(Collectors.toList());
+        final List<AuthorizationAttribute> writePermissions = allPermissions.stream()
+                .filter(p -> p.getOperation() == AuthorizationService.Operation.WRITE)
+                .map(p -> p.getAuthorizationAttribute())
+                .collect(Collectors.toList());
+        return new AdminAuthorization(adminPermissions, readPermissions, writePermissions);
+    }
+
+    public void updateAdmins(final AdminAuthorization newAdmins) throws InsufficientAuthorizationException {
+        if (newAdmins.getAdmins().isEmpty() || newAdmins.getWriters().isEmpty() || newAdmins.getReaders().isEmpty()) {
+            throw new InsufficientAuthorizationException("There must be at least one administrator in each list");
+        }
+        final AdminAuthorization currentAdmins = getAdmins();
+        transactionTemplate.execute(action -> {
```

Comment:
Do you think you could move trasaction in the place where you requests to db?

Reply:
425df1d